app/helpers/view_helper.rb

```diff
@@ -122,6 +122,30 @@ def protect_against_mistakes(anchor:)
     end
   end
 
+  UTM_PARAMS = {
+    utm_source: 'apply-for-teacher-training.service.gov.uk',
+    utm_medium: 'referral',
+  }.freeze
+
+  def govuk_link_to_with_utm_params(link_text, link_url, utm_campaign, utm_content = nil, **extra_options)
+    url = construct_url(link_url, utm_campaign, utm_content)
+    govuk_link_to link_text.to_s, url.to_s, extra_options
+  end
+
+  def email_link_with_utm_params(link_url, utm_campaign, utm_content)
+    url = construct_url(link_url, utm_campaign, utm_content)
+    url.to_s
+  end
+
+  def utm_campaign(path, context = nil)
+    path.sub!(/\/candidate\/application/, '')
```

Comment:
this feels a bit brittle. do you think it would be viable to rely on the controller and action, as opposed to the path?

ie something like `params.slice(:controller, :action).values.join('-')`